ddtrace/tracer/span.go

```diff
@@ -337,6 +340,37 @@ func (s *span) String() string {
 	return strings.Join(lines, "\n")
 }
 
+// Format implements fmt.Formatter.
+func (s *span) Format(f fmt.State, c rune) {
+	switch c {
+	case 's':
+		fmt.Fprint(f, s.String())
+	case 'v':
+		if svc := globalconfig.ServiceName(); svc != "" {
```

Comment:
This might not be set, yet the tracer might have a service which we want to display, set by for example `WithServiceName`. Using `globalconfig.ServiceName()` should be a fallback when a tracer is not available. Did I miss something here?

Reply:
I'm ok with this since `WithServiceName` is deprecated.

We *only* want to report the service name here when it has been explicitly set by the user, and that is exactly when `globalconfig.ServiceName()` has a value, except for when using the deprecated `WithServiceName` option.

Reply:
Even so, why not use the service set in the tracer every time? Regardless of where it is coming from. What if there's no global service but the default is set in the tracer (the binary name) ? Then we want to use that. I think the way to go is:

* Take the service, env, version from the active tracer.

If the tracer is stopped:

* Take the service from globalconfig if it's there
* Take from envs...

Reply:
The way you have it currently is: if there's no active tracer, and no global service, no service will be appended to the log. That's not useful. If you change it the way I'm suggesting, then in this scenario you will get the binary name which will at least mean something to people rather than having logs lingering around without any service attached.

Reply:
> What if there's no global service but the default is set in the tracer (the binary name) ? Then we want to use that.
We actually don't want to use the default (binary name) according to the spec and the conversations on slack. I raised an issue about this several times because I agree that it would be better to use the binary name, but that wasn't the resolution.

>The way you have it currently is: if there's no active tracer, and no global service, no service will be appended to the log.
This is by design, and what was asked for.

> That's not useful. [...] Let's never have an empty service. In a big infrastructure those are completely lost logs.
I 100% agree. Let's raise this issue again and see if we can get the spec changed.

Reply:
We resolved this elsewhere, I think. Let me know if there is anything else we need to do.